src/exchange/exchange_client.rs

```diff
@@ -867,6 +905,176 @@ impl ExchangeClient {
 
         self.post(action, signature, timestamp).await
     }
+
+    // Multi-sig methods
+
+    async fn post_multi_sig(
+        &self,
+        action: serde_json::Value,
+        signatures: Vec<Signature>,
+        nonce: u64,
+    ) -> Result<ExchangeResponseStatus> {
+        let exchange_payload = MultiSigExchangePayload {
+            action,
+            signatures,
+            nonce,
+            vault_address: self.vault_address,
+        };
+        let res = serde_json::to_string(&exchange_payload)
+            .map_err(|e| Error::JsonParse(e.to_string()))?;
+        debug!("Sending multi-sig request {res:?}");
+
+        let output = &self
+            .http_client
+            .post("/exchange", res)
+            .await
+            .map_err(|e| Error::JsonParse(e.to_string()))?;
+        debug!("Response: {output}");
+        serde_json::from_str(output).map_err(|e| Error::JsonParse(e.to_string()))
+    }
```

Comment:
I don't think you are creating this correctly @nicolad 

https://github.com/hyperliquid-dex/hyperliquid-python-sdk/blob/8baad667965968a020a2bb90d0287df0922ca941/hyperliquid/exchange.py#L1068-L1093

Reply:
@nicolad I'll actually try it, but looks like it will work if you have access to all the signers which is fine in my case. But this isn't something you can assume in most multisig cases. I think you'd want the function to take Signatures and not PrivateKeySigners? Also might need a way to export a signature. I'd think the usual flow is you collect signatures from the other wallets and when you have enough you as the submitter add your last signature and attach all the other signatures you collected.


```
pub async fn multi_sig_usdc_transfer(
        &self,
        multi_sig_user: Address,
        amount: &str,
        destination: &str,
        wallets: &[PrivateKeySigner],
    )
```